src/main/java/io/github/milkitic/minecraft/plugin/UserStatistic.java

```diff
@@ -54,8 +53,10 @@
     @ConfigDir(sharedRoot = false)
     private Path configDir;
 
-    private HashMap<String, Long> _totalSecondMap;
-    private HashMap<String, Calendar> _currentDateMap;
+    private Map<String, Long> _totalSecondMap;
+    private Map<String, Calendar> _currentDateMap; //login date
+
+    public static Calendar lastUpdated;
```

Comment:
Our static analysis engine detects that your pull request includes a change of defining a static `Calendar` field. According to [this description](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#stcal-static-calendar-field-stcal-static-calendar-instance), `Calendar` is inherently unsafe for multithreaded use, and sharing a single instance across thread boundaries without proper synchronization will result in erratic behavior of the application.